src/locale/keymap-util.c

```diff
@@ -780,3 +781,137 @@ int x11_convert_to_vconsole(Context *c) {
 
         return modified;
 }
+
+bool locale_gen_check_available(void) {
+        struct stat st;
+        if (stat("/usr/sbin/locale-gen", &st) < 0)
+                return false;
+        if (stat("/etc/locale.gen", &st) < 0)
+                return false;
+        return true;
+}
+
+int locale_gen_enable_locale(char *locale[_VARIABLE_LC_MAX]) {
+        _cleanup_fclose_ FILE *fr = NULL;
+        _cleanup_fclose_ FILE *fw = NULL;
+        _cleanup_free_ char *temp_path = NULL;
+        _cleanup_free_ char *locale_entry = NULL;
+        bool locale_enabled = false;
+        bool first_line;
+        struct stat st;
+        int r;
+        const char *LOCALE_GEN_CONF_FNAME = "/etc/locale.gen";
```

Comment:
i don#t think there's benefit of having this variable. First of all it allocates a redundant pointer on stack, but most importantly it's not really a useful variable since it doesn't hide or abstract anything. It's value is always the same, and it's value is almost literally the same as its name. 

Defining constants that abstract values makes a lot of sense when the value can change, or when the value is some arbitrarily chosen value, or if the value is really hard to type and make mistakes in. but if neither is the case, it's much better and simpler to just use the literal value. i.e. just pass the literal string "/etc/locale.gen" to fopen() please.

Reply:
For me it was more a "only one place to introduce a hard-to-spot typo" kind of thing. But it actually isn't even amazing for that purpose the way I used it here, as I didn't define this as a global #define but only locally. So, literals it is :-)
The config file path is pretty much guaranteed to not change - it hasn't in the last decade(s).